cmake/AddSourcesFunctions.cmake

```diff
@@ -0,0 +1,43 @@
+function(add_sources)
+   foreach(_source IN ITEMS ${ARGN})
+       if (IS_ABSOLUTE "${_source}")
+           set(source_abs "${_source}")
```

Comment:
Did you mean to say `_source_abs`? Otherwise a file with absolute path doesn't get appended to `GlobalSourceList`